### ISSUE TITLE
torch_geometric.datasets.word_net: retrieve synsets and lemmas

### ISSUE THREAD

**Issue by matteomedioli** [pyg-team/pytorch_geometric#1991]
## ❓ Questions & Help

I was wondering if there is a way to get more data from the WordNet18 dataset (https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/datasets/word_net.html#WordNet18). If not, I will try to create my own dataset.
 
About edges, what the library returns is a graph of this kind:
_(Data(edge_index=[2, 141442], edge_type=[141442]), {'edge_index': tensor([0, 141442]), 'edge_type': tensor([0, 141442])})_
(a graph described by 141442 triplets <node1, node2, relation> where however the labels associated with the relations are considered all different, so 141442 edge type?) There's no relations between this edges and relations described in WN (hypernym, hyponym, meronym etc)?

About nodes: in this implementation of WordNet18 (both in gpickle file than returned Data)  nodes are described just by a nodeID. Is there a way to retrieve synsets or lemmas from this nodeIDs?

Thank you for your amazing work.

**Comment by rusty1s** [comment]
You are right that each edge is denoted by an individual edge type, but those relations are not considered all different. In fact, there are exactly 18 different edge relations:
```python
data.edge_type.min() == 0
data.edge_type.max() == 17
```
If I'm not mistaken, the indices should correspond to the [following meaning](https://github.com/villmow/datasets_knowledge_embedding/blob/master/WN18/original/relation2id.txt):
```
_member_of_domain_topic	0
_member_meronym	1
_derivationally_related_form	2
_member_of_domain_region	3
_similar_to	4
_hypernym	5
_member_holonym	6
_instance_hypernym	7
_member_of_domain_usage	8
_synset_domain_topic_of	9
_hyponym	10
_instance_hyponym	11
_synset_domain_usage_of	12
_has_part	13
_verb_group	14
_part_of	15
_synset_domain_region_of	16
_also_see	17
```

**Comment by matteomedioli** [comment]
> You are right that each edge is denoted by an individual edge type, but those relations are not considered all different. In fact, there are exactly 18 different edge relations:
> 
> ```python
> data.edge_type.min() == 0
> data.edge_type.max() == 17
> ```
> 
> If I'm not mistaken, the indices should correspond to the [following meaning](https://github.com/villmow/datasets_knowledge_embedding/blob/master/WN18/original/relation2id.txt):

Thank you very much! I was looking for exactly these two files! 
I've a little doubt about entity2id file: are the ids equivalent to offsets? If a synset it is uniquely represented by a pair <pos, offset>, how I can retrieve a node related entity from this file only using offset?
For example id (offset) 1257145 represents both sensual.s.02 than precession.n.02. Maybe I didn't fully understand the meaning of the ids. Thanks in advance

**Comment by rusty1s** [comment]
`entity2id`just maps the entity number to a contiguous id. The actual graph structure is located in the files `train.txt`, `valid.txt`, and so on.
